File: php/src/Exception/ApiException.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Exception;

/** API error with HTTP status and response details. */
class ApiException extends DocGenException
{
    public function __construct(
        public readonly int $statusCode,
        string $message,
        public readonly ?string $responseBody = null,
    ) {
        parent::__construct($message, $statusCode);
    }
}

File: java/src/main/java/com/dokmatiq/docgen/exception/ApiException.java
package com.dokmatiq.docgen.exception;

/** API error with HTTP status and response details. */
public class ApiException extends DocGenException {
    private final int statusCode;
    private final String responseBody;

    public ApiException(int statusCode, String message, String responseBody) {
        super(message);
        this.statusCode = statusCode;
        this.responseBody = responseBody;
    }

    public int statusCode() { return statusCode; }
    public String responseBody() { return responseBody; }
}
